geometric_kernels/spaces/so.py

```diff
@@ -0,0 +1,368 @@
+            partition[1] += 1
+            continue
+        j = 2
+        s = partition[0] + partition[1] - 1
+        while j < L and partition[j] >= partition[0] - 1:
+            s += partition[j]
+            j += 1
+        if j >= L:
+            return
+        partition[j] = x = partition[j] + 1
+        j -= 1
+        while j > 0:
+            partition[j] = x
+            s -= x
+            j -= 1
+        partition[0] = s
+
+
+def partition_dominance_cone(partition):
+    '''
+    Calculates partitions dominated by a given one
+    and having the same number of parts (including zero parts of the original)
+    '''
+    cone = {partition}
+    new_partitions = {0}
+    prev_partitions = cone
+    while new_partitions:
+        new_partitions = set()
+        for partition in prev_partitions:
+            for i in range(len(partition) - 1):
+                if partition[i] > partition[i + 1]:
+                    for j in range(i + 1, len(partition)):
+                        if partition[i] > partition[j] + 1 and partition[j] < partition[j - 1]:
+                            new_partition = list(partition)
+                            new_partition[i] -= 1
+                            new_partition[j] += 1
+                            new_partition = tuple(new_partition)
+                            if new_partition not in cone:
+                                new_partitions.add(new_partition)
+        cone.update(new_partitions)
+        prev_partitions = new_partitions
+    return cone
+
+
+def partition_dominance_or_subpartition_cone(partition):
+    '''
+        Calculates subpartitions and partitions dominated by a given one
+        and having the same number of parts (including zero parts of the original)
+        '''
+    cone = {partition}
+    new_partitions = {0}
+    prev_partitions = cone
+    while new_partitions:
+        new_partitions = set()
+        for partition in prev_partitions:
+            for i in range(len(partition) - 1):
+                if partition[i] > partition[i + 1]:
+                    new_partition = list(partition)
+                    new_partition[i] -= 1
+                    new_partition = tuple(new_partition)
+                    if new_partition not in cone:
+                        new_partitions.add(new_partition)
+                    for j in range(i + 1, len(partition)):
+                        if partition[i] > partition[j] + 1 and partition[j] < partition[j - 1]:
+                            new_partition = list(partition)
+                            new_partition[i] -= 1
+                            new_partition[j] += 1
+                            new_partition = tuple(new_partition)
+                            if new_partition not in cone:
+                                new_partitions.add(new_partition)
+        cone.update(new_partitions)
+        prev_partitions = new_partitions
+    return cone
+
+
+class SOEigenfunction(LieGroupAddtitionTheorem):
```

Comment:
We usually name this kind of classes in the plural: `Eigenfunctions`, not `Eigenfunction`. I think it is worth sticking to this tradition